### PR TITLE
Replaces result listener animation with a Glide animation

### PR DESCRIPTION
**Pull request by antonis**
Partially #12629

## Description
As discussed in https://github.com/wordpress-mobile/WordPress-Android/pull/20409 there are many roots for this issue that require a wider refactoring. While debugging the issue I noticed that removing the animation completely fixed the bug for me. Thus I tried an alternative approach by replacing the custom result listener animation with a Glide animation. Note that the animation resource used is the same and the experience should not change for the user.

-----

## To Test:
1. Uninstall previous versions of the app or purge the app cache
2. Install the Jetpack app from this branch
3. Login with an account that has achievements
4. Switch to the notification tab
5. Open an achievement
6. **Verify** that the image loads

### To reproduce:
I was able to reproduce the issue on `trunk` consistently with the steps above unrelated to the day/night mode (see https://github.com/wordpress-mobile/WordPress-Android/issues/12629#issuecomment-1976994225)

-----

## Regression Notes

1. Potential unintended areas of impact

    - Achievement notification details screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

3. What automated tests I added (or what prevented me from doing so)

    - Added test cases

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>GlidePopTransition</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>GlidePopTransitionFactory</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/util/image/GlidePopTransitionOptions.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.bumptech.glide.TransitionOptions
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.request.transition.NoTransition
+import com.bumptech.glide.request.transition.Transition
+import com.bumptech.glide.request.transition.TransitionFactory
+import org.wordpress.android.R
+import org.wordpress.android.util.AppLog
+import java.lang.RuntimeException
+
+object GlidePopTransitionOptions : TransitionOptions<GlidePopTransitionOptions, Drawable>() {
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/util/image/GlidePopTransitionOptions.kt:14]: I used [DrawableTransitionOptions](https://github.com/bumptech/glide/blob/master/library/src/main/java/com/bumptech/glide/load/resource/drawable/DrawableTransitionOptions.java) as a guide for this implementation.

**Comment by mkevins** [comment]
The code changes look good, and I like that it simplifies things to use the Glide animation. But, for whatever reason, I'm still getting a blank page on the first load from the welcome notification:



I also verified via the network inspector that the image response on the network was received correctly (with a status code `200`), so it must be something about our view logic that is somehow preventing it actually loading. I will dig further to see if I can find any clues about why it sometimes fails to load.

**Comment by mkevins** [comment]
I added some more logs to try to understand what is happening with the image resource (since the glide logic must be invoked at somepoint, otherwise the network inspector wouldn't show the successful request and response).

<details>
<summary>logs patch</summary>

```diff
diff --git a/WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/NoteBlock.java b/WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/NoteBlock.java
index 7c453730f89..1ecb8dc71a7 100644
--- a/WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/NoteBlock.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/NoteBlock.java
@@ -6,6 +6,7 @@ import android.net.Uri;
 import android.text.Spannable;
 import android.text.TextUtils;
 import android.text.style.TypefaceSpan;
+import android.util.Log;
 import android.view.Gravity;
 import android.view.View;
 import android.view.ViewGroup;
@@ -156,6 +157,7 @@ public class NoteBlock {
 
         // Note image
         if (hasImageMediaItem()) {
+            Log.d("snaplog", "has image media item");
             noteBlockHolder.getImageView().setVisibility(View.VISIBLE);
             // Request image, and animate it when loaded
             mImageManager.animateWithResultListener(noteBlockHolder.getImageView(), ImageType.IMAGE,
@@ -164,6 +166,7 @@ public class NoteBlock {
                     new ImageManager.RequestListener<Drawable>() {
                         @Override
                         public void onLoadFailed(@Nullable Exception e, @Nullable Object model) {
+                            Log.d("snaplog", "load failed");
                             if (e != null) {
                                 AppLog.e(T.NOTIFS, e);
                             }
@@ -176,18 +179,22 @@ public class NoteBlock {
                     });
 
             if (mIsBadge) {
+                Log.d("snaplog", "is badge, so setting important for accessibility");
                 noteBlockHolder.getImageView().setImportantForAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_NO);
             }
         } else {
+            Log.d("snaplog", "doesn't have image media item");
             mImageManager.cancelRequestAndClearImageView(noteBlockHolder.getImageView());
             noteBlockHolder.hideImageView();
         }
 
         // Note video
         if (hasVideoMediaItem()) {
+            Log.d("snaplog", "has video media item");
             noteBlockHolder.getVideoView().setVideoURI(Uri.parse(StringUtils.notNullStr(getNoteMediaItem().getUrl())));
             noteBlockHolder.getVideoView().setVisibility(View.VISIBLE);
         } else {
+            Log.d("snaplog", "doesn't have video media item");
             noteBlockHolder.hideVideoView();
         }
 
@@ -310,6 +317,7 @@ public class NoteBlock {
 
         public ImageView getImageView() {
             if (mImageView == null) {
+                Log.d("snaplog", "image view is null ... creating a new one");
                 mImageView = new ImageView(mRootLayout.getContext());
                 int imageSize = DisplayUtils.dpToPx(mRootLayout.getContext(), 180);
                 int imagePadding = mRootLayout.getContext().getResources().getDimensionPixelSize(R.dimen.margin_large);
diff --git a/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt b/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt
index 591bb748cf8..91804ceaf26 100644
--- a/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt
@@ -11,6 +11,7 @@ import android.graphics.drawable.Drawable
 import android.net.Uri
 import android.text.TextUtils
 import android.util.Base64
+import android.util.Log
 import android.widget.ImageView
 import android.widget.ImageView.ScaleType
 import android.widget.ImageView.ScaleType.CENTER
@@ -314,7 +315,10 @@ class ImageManager @Inject constructor(
         requestListener: RequestListener<Drawable>
     ) {
         val context = imageView.context
-        if (!context.isAvailable()) return
+        if (!context.isAvailable()) {
+            Log.d("snaplog", "context not available for glide load")
+            return
+        }
         Glide.with(context)
             .load(Uri.parse(imgUrl))
             .addFallback(imageType)
```
</details>

<details>
<summary>logs</summary>

```
2024-03-08 18:11:26.601 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.601 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.604 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.604 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.610 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.612 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.618 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.618 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.622 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.623 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.632 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.632 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.633 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.633 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.639 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.640 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.640 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.643 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.644 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.648 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.649 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.657 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.658 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.665 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.666 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.666 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.667 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.696 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.696 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.698 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.698 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.699 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.700 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.705 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.705 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.710 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.710 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.716 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.717 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.717 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.718 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.718 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.723 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.723 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.727 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.727 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.734 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.735 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.735 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.762 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.762 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.762 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.791 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.792 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.792 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.822 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.822 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.822 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.852 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.852 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.852 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.876 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.876 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.876 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.886 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:26.888 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:26.888 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.914 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.914 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.915 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.941 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.941 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.942 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:26.968 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:26.968 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:26.969 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.168 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:27.168 10105-10105 snaplog                 com.jetpack.android.prealpha         D  image view is null ... creating a new one
2024-03-08 18:11:27.170 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:27.170 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.171 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.171 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.175 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.176 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.179 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.180 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.183 10105-10105 snaplog                 com.jetpack.android.prealpha         D  has image media item
2024-03-08 18:11:27.184 10105-10105 snaplog                 com.jetpack.android.prealpha         D  is badge, so setting important for accessibility
2024-03-08 18:11:27.184 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.185 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.185 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.189 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.190 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
2024-03-08 18:11:27.193 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have image media item
2024-03-08 18:11:27.193 10105-10105 snaplog                 com.jetpack.android.prealpha         D  doesn't have video media item
```
</details>

Maybe the most suspicious thing so far is the number of times that `mImageView` is being created new. It seems that `mImageView` and `mVideoView` are "special" in the existing implementation, in that they are created programmatically, and inserted into the view hierarchy. I'm not confident that this code is thread-safe, so this feels like it could be a culprit in race conditions.

**Comment by antonis** [comment]
Thank you for testing @mkevins 🙇 

>  so it must be something about our view logic that is somehow preventing it actually loading.

Yes, I guess there is some race condition that prevents rendering in some occasions :(

> But, for whatever reason, I'm still getting a blank page on the first load from the welcome notification:

Given that the issue is now not reproducible on my test device (Pixel 5, Android 14) the rendering might fail on specific devices due to performance or network conditions 🤔 

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/abe0cd56-b1b3-4cfc-bc8f-66c90a4539c3

What device are you testing on?

**Comment by mkevins** [comment]
> What device are you testing on?

I have been testing on a Pixel 3a (physical device), and the network load times could indeed be a factor (mine is currently fairly slow). But, what is slightly concerning to me is that the network inspector showed a 200 response, and yet the image still did not load.. so it's not just a matter of timing out, or waiting long enough for the image to come over the wire. I'm confident that there is something broken in the view logic itself 😞 .

**Comment by antonis** [comment]
> It seems that mImageView and mVideoView are "special" in the existing implementation, in that they are created programmatically, and inserted into the view hierarchy. I'm not confident that this code is thread-safe, so this feels like it could be a culprit in race conditions.

Sounds possible 👍 The weird thing is that this doesn't always happen. I was able to reproduce it consistently only with bigger images (that's why I was tricked to believe that [the x1 solution](https://github.com/wordpress-mobile/WordPress-Android/pull/20409) would work). So maybe when the image loading is delayed something happens that prevents rendering 🤔

**Comment by antonis** [comment]
> I added some more logs to try to understand what is happening with the image resource (since the glide logic must be invoked at somepoint, otherwise the network inspector wouldn't show the successful request and response).

<details>
<summary>Applied the same patch on my device that is rendering the image to compare the logs</summary>

```
2024-03-08 10:49:20.971  7048-7048  snaplog                 com.jetpack.android.beta             D  has image media item
2024-03-08 10:49:20.971  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:20.973  7048-7048  snaplog                 com.jetpack.android.beta             D  is badge, so setting important for accessibility
2024-03-08 10:49:20.973  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.976  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.977  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.981  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.981  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.983  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.984  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.990  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.991  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.991  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.991  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.995  7048-7048  snaplog                 com.jetpack.android.beta             D  has image media item
2024-03-08 10:49:20.996  7048-7048  snaplog                 com.jetpack.android.beta             D  is badge, so setting important for accessibility
2024-03-08 10:49:20.996  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:20.998  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:20.998  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.001  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.001  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.003  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.004  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.008  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.008  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.009  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.009  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.013  7048-7048  snaplog                 com.jetpack.android.beta             D  has image media item
2024-03-08 10:49:21.013  7048-7048  snaplog                 com.jetpack.android.beta             D  is badge, so setting important for accessibility
2024-03-08 10:49:21.013  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.029  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.029  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.029  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.045  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.045  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.045  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.059  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.059  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.059  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.075  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.075  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.076  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.088  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.088  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.089  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.564  7048-7048  snaplog                 com.jetpack.android.beta             D  has image media item
2024-03-08 10:49:21.564  7048-7048  snaplog                 com.jetpack.android.beta             D  image view is null ... creating a new one
2024-03-08 10:49:21.565  7048-7048  snaplog                 com.jetpack.android.beta             D  is badge, so setting important for accessibility
2024-03-08 10:49:21.565  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.567  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.568  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.571  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.571  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.573  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.573  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.579  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.579  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.580  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.580  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.584  7048-7048  snaplog                 com.jetpack.android.beta             D  has image media item
2024-03-08 10:49:21.584  7048-7048  snaplog                 com.jetpack.android.beta             D  is badge, so setting important for accessibility
2024-03-08 10:49:21.584  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.586  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.587  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.589  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.590  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.592  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.592  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.597  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.597  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
2024-03-08 10:49:21.598  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have image media item
2024-03-08 10:49:21.598  7048-7048  snaplog                 com.jetpack.android.beta             D  doesn't have video media item
```

</details>

**Comment by antonis** [comment]
Comparing the two logs the differences are the following

Screenshot 2024-03-08 at 10 58 45 AM
Screenshot 2024-03-08 at 10 58 03 AM

**Comment by antonis** [comment]
Hey @mkevins 👋 
I made another attempt to address this issue by changing the ImageView initialisation with 44b79ababe0d4ff2511af816cba6bb4186e7777e. If you have some time I would appreciate retesting since I'm unable to reproduce on my side 🙇

**Comment by mkevins** [comment]
>  If you have some time I would appreciate retesting since I'm unable to reproduce on my side 

Sadly, the same result occurred: a blank / missing image on the first load. 😞

**Comment by antonis** [comment]
I made a few more attempts to fix this but without a success. The good thing is that I am now able to reproduce this (almost) consistently on my Pixel 5 (Android 14) device after a sign up. The weird thing is that when I clear all app caches (or even delete the app and reinstall) I'm not able to reproduce the issue with the same account 🤔 
This makes me believe that despite the issues we have identified with the code structure the issue might be attributed to a delay on fetching the image from the server on a new account.

https://github.com/wordpress-mobile/WordPress-Android/assets/304044/72935c1b-8c10-4a41-af86-bf46f395cbe6

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/279">24.5</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/279">24.5</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by antonis** [comment]
> This makes me believe that despite the issues we have identified with the code structure the issue might be attributed to a delay on fetching the image from the server on a new account.

With this assumption I added image preloading with ab029691d5fab6b72b4bdc58538ecb118a4e43a5

@mkevins If you have the time please check if that fixes the issue for you 🙇

**Comment by mkevins** [comment]
> I added image preloading

Good thinking!

> If you have the time please check if that fixes the issue for you 

I re-tested this, and I can no longer reproduce the missing image on the welcome notification! 🎉

**Review by mkevins — APPROVED** [review]
Thank you for this improvement Antonis! With the latest changes (the preloading), this resolves the missing welcome image in my testing. Code changes look good too, and thank you also for adding tests. 👍

**Comment by antonis** [comment]
Thank you for reviewing and testing again @mkevins 🙇 :)

**Pull request merged**